### PR TITLE
[Impeller] give geometry classes similar API as entity / implement strokepathgeom

### PR DESCRIPTION
**Pull request by jonahwilliams**
While working on making a StrokePathGeometry, I found that I needed to update the API again to provide more data for the smoothing approximation: https://github.com/flutter/engine/blob/main/impeller/entity/contents/solid_stroke_contents.cc#L213

Rather than pass through each object we need one at a time, lets just pass the same context that the Entity::Render method gets, at least for now.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Converting to a draft for now, I pushed up what the changes would look like with (most of) SolidStrokeGeometry implemented

File: impeller/entity/contents/solid_stroke_contents.cc
Comment (jonahwilliams) [inline impeller/entity/contents/solid_stroke_contents.cc:63]: I think this is the right check :)

Once we merge solid fill + stroke, we'll need to retain these clips/options I'm assuming.

File: impeller/entity/geometry.h
Comment (bdero) [inline impeller/entity/geometry.h:66]: Passing around the `ContentContext` & `Entity` SGTM. The filter graph started off with minimal context being passed around and quickly ended up in a similar place.

**Review by bdero — APPROVED** [review]
Great! This LGTM assuming `AiksTest.SolidStrokesRenderCorrectly` and `EntityTest.StrokeCapAndJoinTest` are looking good.

**Comment by jonahwilliams** [comment]
I've verified everything is working correctly on those tests and updated to ToT. Going to land this now so we can close out this refactor. If anyone notices a problem with strokes though, this PR is the culprit 😆

**Pull request merged**
